engine/src/test/java/io/zeebe/engine/util/TestStreams.java

```diff
@@ -263,6 +263,11 @@ private StreamProcessor buildStreamProcessor(
 
     final ActorFuture<Void> openFuture = new CompletableActorFuture<>();
 
+    try {
+      currentSnapshotController.recover();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
I guess that this is only necessary if we deleted the snapshots before. If so, we could also do it in `EngineRule#reprocess()`.